tests/test_hdfs.py

```diff
@@ -125,3 +128,65 @@ def test_terasort_suite():
                     expected_output="partitions are properly sorted",
                     app_name="/spark",
                     args=teravalidate_args)
+
+
+@pytest.mark.skipif(not utils.hdfs_enabled(), reason='HDFS_ENABLED is false')
+@pytest.mark.sanity
+def test_supervise():
```

Comment:
There's already a test with the same name in test_spark.py. Did you want to rename it to something like test_supervise_with_checkpointing?

Reply:
ah no, I should remove the other test.